envoy/config/filter/accesslog/v2/accesslog.proto

```diff
@@ -414,6 +415,44 @@ message OrFilter {
   repeated AccessLogFilter filters = 2 [(validate.rules).repeated .min_items = 2];
 }
 
+// [#not-implemented-hide:] Filters requests based on the presence or value of a request header.
+message HeaderValueFilter {
```

Comment:
I know that `HeaderMatcher` has some deprecated stuff in it, but can we just use that object directly here for code sharing? Then we can just deprecate in all places at once eventually. You could just embed it inside a filter message?

Reply:
Makes sense, will do.